Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java

```diff
@@ -549,27 +549,33 @@ public void onPlayerCommandPreprocess(final PlayerCommandPreprocessEvent event)
     @EventHandler(priority = EventPriority.NORMAL)
     public void onPlayerChangedWorldFlyReset(final PlayerChangedWorldEvent event) {
         final User user = ess.getUser(event.getPlayer());
-        if (user.getBase().getGameMode() != GameMode.CREATIVE
-                // COMPAT: String compare for 1.7.10
-            && !user.getBase().getGameMode().name().equals("SPECTATOR")
-            && !user.isAuthorized("essentials.fly")) {
-            user.getBase().setFallDistance(0f);
-            user.getBase().setAllowFlight(false);
-        }
-        if (!user.isAuthorized("essentials.speed")) {
-            user.getBase().setFlySpeed(0.1f);
-            user.getBase().setWalkSpeed(0.2f);
-        } else {
-            if (user.getBase().getFlySpeed() > ess.getSettings().getMaxFlySpeed() && !user.isAuthorized("essentials.speed.bypass")) {
-                user.getBase().setFlySpeed((float) ess.getSettings().getMaxFlySpeed());
-            } else {
-                user.getBase().setFlySpeed(user.getBase().getFlySpeed() * 0.99999f);
+
+        if (ess.getSettings().isWorldChangeFlyResetEnabled()) {
```

Comment:
This doesn't need to be nested.

Reply:
No problem, done, https://github.com/EssentialsX/Essentials/pull/2546/commits/36d620129e798c5e44a8e8ffa188b8294be9233d

This was just in the original code so I thought this was fine.